### DISCUSSION TITLE
New Keycloak Admin UI

### DISCUSSION THREAD

**Discussion by ssilvert** [keycloak/keycloak#9119] [Feedback]
The UI Team is very excited that the New Admin UI is now available as an experimental feature in Keycloak 15.1.0.  Right now, it is about 90% feature complete.

So we are now at the point we want to solicit feedback from the community. 

To enable the UI:

1. Pass `-Dkeycloak.profile.feature.admin2=enabled` at startup
2. Go to `Realm Settings --> Themes`
3. Set the admin theme to `keycloak.v2`
4. Navigate to `http://<host and port>/auth/admin/master/console/`

For any issues you find, please report to the issue tracking system in our dedicated GitHub Repo: https://github.com/keycloak/keycloak-admin-ui/issues.

Thanks in advance,

Stan

**Comment by chrisx8** [comment]
I can't seem to enable the new Admin UI in Keycloak.X. 

I started the Docker container with `docker run -p 8080:8080 -e KEYCLOAK_ADMIN=init -e KEYCLOAK_ADMIN_PASSWORD=*** quay.io/keycloak/keycloak-x:15.1.0 start-dev -Dkeycloak.profile.feature.admin2=enabled`. The output in the terminal shows `admin2` as enabled, but when I log in to the admin console, I only see `base` and `keycloak` under `Realm Settings --> Themes -> Admin Console Theme`. `keycloak.v2` isn't an option there.

See #9120 for more details.

**Reply by pedroigor** [reply]
In Dist.X we do not have themes exploded within the distribution but loaded from classpath by default.

Looks like we are missing the `admin` here https://github.com/keycloak/keycloak/blob/0afd55f32c8f833dba9958cb3dd7c297dccb3edc/themes/src/main/resources-community/META-INF/keycloak-themes.json#L10.

I'm sending a PR to enable the new admin.

**Reply by pedroigor** [reply]
See https://github.com/keycloak/keycloak/issues/9128

**Comment by dasniko** [comment]
I just had a quick look into it.
Ok, it's... different, perhaps somewhat more modern (whatever that means). I don't know if it's really "better", IMO. I'm used to the old layout for 7 years now, and for sure I will get used to the new one, after some time.

I discovered several smaller issues:

I run my server with `-Dkeycloak.profile=preview -Dkeycloak.profile.feature.admin2=enabled`, but UPLOAD_SCRIPTS and DOCKER features are both enabled and disabled?
![image 2021-12-16 um 14 04 04](https://user-images.githubusercontent.com/2075748/146379108-1887fe0b-0a1a-4049-a6b0-3b80047f30c4.png)

In client credentials tab, the default setting is still "client-id and secret" and the secret is shown, but the authenticator field shows "signed jwt" as it is the first option in the dropdown, not the selected one.
![image 2021-12-16 um 14 06 52](https://user-images.githubusercontent.com/2075748/146379402-df22f4ae-d6ba-4054-8853-60846bde460c.png)

Some of the social IdP icons are missing:
![image 2021-12-16 um 14 14 07](https://user-images.githubusercontent.com/2075748/146379564-e5fb9013-79ea-46bf-89ef-28fd12e47718.png)

If there are no user fed providers configured, I can't choose my custom extension from a list, only the built-ins are available.
![image 2021-12-16 um 14 15 01](https://user-images.githubusercontent.com/2075748/146379703-be3098ef-dae3-4169-9d48-ae21b2d4347e.png)
In a realm with a configured custom user provider, I can see it and also have a selection field:
![image 2021-12-16 um 14 15 21](https://user-images.githubusercontent.com/2075748/146379808-91df4342-069a-407e-9806-bfb1e3d10dd2.png)

All over the various screens, some labels seem to be "too long", as they are truncated/overlayed by fields.
![image 2021-12-16 um 14 08 02](https://user-images.githubusercontent.com/2075748/146379502-174c6fa5-2f2d-4137-a026-d7d03dffe535.png)

That's it what I noticed at a first shot.
Hope that it helps.

**Comment by ssilvert** [comment]
Thanks @dasniko.  That was very helpful.

I'll see that your feedback gets logged as issues.  For anything else you find, you can create your own issues in our database at  https://github.com/keycloak/keycloak-admin-ui/issues.

**Reply by dasniko** [reply]
Will do in future.
After I wrote the post, I saw the separate repo and was too lazy to dig through all the issues if my findings were already there. Blame on me. 🙃

**Reply by ssilvert** [reply]
It's OK.  I just finished adding your issues.  Thanks again.
